relay-server/src/utils/dynamic_sampling.rs

```diff
@@ -53,6 +79,31 @@ fn get_sampling_match(
     )
 }
 
+/// Checks whether an incoming event should be kept or dropped based only on trace rules, thus
+/// without requiring the configuration of the non-root project.
+///
+/// This method has been duplicated from the `should_keep_event` since it should be used independently
+/// and a clear separation is needed.
+pub fn should_keep_event_with_trace_rules(
+    processing_enabled: bool,
+    root_project_state: Option<&ProjectState>,
+    dsc: Option<&DynamicSamplingContext>,
+    ip_addr: Option<IpAddr>,
+) -> SamplingResult {
+    let sampling_result = get_sampling_match_result(
+        processing_enabled,
+        None,
+        root_project_state,
+        dsc,
+        None,
+        ip_addr,
+        // For consistency reasons we take a snapshot in time and use that time across all code that
+        // requires it.
+        Utc::now(),
+    );
+    SamplingResult::determine_from_sampling_match(sampling_result)
+}
```

Comment:
Regardless of what we discussed today, I would remove this function and just adapt the interface of `should_keep_event` (and inline `determine_from_sampling_match` again) to reduce code duplication. We should maybe rename `should_keep_event` to `get_sampling_decision` or similar now that it has become so generic.